app/controllers/v1/sessions_controller.rb

```diff
@@ -74,28 +73,20 @@ def metadata
 
     private
 
-    # rubocop:disable Metrics/CyclomaticComplexity
     def redirect_url(type)
+      raise Common::Exceptions::RoutingError, type unless REDIRECT_URLS.include?(type)
+
       case type
-      when 'signup'
-        url_service.signup_url
-      when 'mhv'
-        url_service.mhv_url
-      when 'dslogon'
-        url_service.dslogon_url
-      when 'idme'
-        url_service.idme_url
-      when 'mfa'
-        url_service.mfa_url
-      when 'verify'
-        url_service.verify_url
+      when 'custom'
+        raise Common::Exceptions::ParameterMissing, 'authn' if params[:authn].blank?
+
+        url_service.custom_url params[:authn]
       when 'slo'
         url_service.ssoe_slo_url # due to shared url service implementation
       else
-        raise Common::Exceptions::RoutingError, params[:path]
+        url_service.public_send("#{type}_url")
```

Comment:
This will _probably_ throw a static analyzer error in CI because you're invoking public_send based on user input (even though user input is constrained by that routing error check above)

Reply:
I refactored this method because CI told me the complexity, or maybe line length, was too much with the addition of the "custom" case statement.  Right now CI is passing, but can you think of another way of doing this?

Reply:
I think the best thing to do would have a heavy-duty refactoring of UrlService, since this method is already wrapping calls to that object anyway.  But that's out-of-scope for what's going on here and for what it's worth, I think it's clearer to have the rubocop exception and a giant case-statement than to use metaprogramming.